homeassistant/components/homekit_controller/__init__.py

```diff
@@ -255,19 +255,23 @@ def update(self):
         # pylint: disable=import-error
         from homekit.exceptions import AccessoryDisconnectedError
 
+        pairing = self._accessory.pairing
+
         try:
-            pairing = self._accessory.pairing
-            data = pairing.list_accessories_and_characteristics()
+            new_values_dict = pairing.get_characteristics(self._chars_to_poll)
```

Comment:
What is more lightweight about this method compared to `list_accessories_and_characteristics`?

Reply:
Why do we not "need to poll the serial number of an accessory once a minute"? Do you mean that, that is what `list_accessories_and_characteristics' does`?

Reply:
Yes exactly. That old api fetches every characteristic on the hub, even ones unrelated to the current entity, and includes schema information and every value the device exports even if it’s something that never changes or we don’t know how to interpret. That includes serial number and manufacturer.

We fetch that payload once a minute.

On BLE this is a nightmare because each characteristic is multiple GATT BLE radio round trips, when we a switch only has 2 characteristics we want to know about.

Reply:
Whereas the new API you give it the characteristic ifs to fetch so can do the minimum amount of radio interactions. It also doesn’t fetch the schema data on every poll either.

Reply:
:+1: that sounds like a good change indeed.